Traceback (most recent call last):
  File "/Users/tyler/.mambaforge/envs/nlp_workshop/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/tyler/.mambaforge/envs/nlp_workshop/lib/python3.11/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tyler/.mambaforge/envs/nlp_workshop/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 166, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tyler/.mambaforge/envs/nlp_workshop/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/tyler/.mambaforge/envs/nlp_workshop/lib/python3.11/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/Users/tyler/.mambaforge/envs/nlp_workshop/lib/python3.11/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/tyler/.mambaforge/envs/nlp_workshop/lib/python3.11/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
def sim_plot(data, hue = None, labels = None, n_colors = 3):
    """Create a scatterplot and optionally color/label its points."""
    fig, ax = plt.subplots(figsize = (10, 10))
    pal = sns.color_palette('colorblind', n_colors = n_colors) if hue else None
    g = sns.scatterplot(
        x = 'x', y = 'y'
        , hue = hue, palette = pal, alpha = 0.8
        , data = data, ax = ax
    )
    g.despine()

    if labels:
        to_label = data[data['label'].isin(labels)]
        to_label[['x', 'y', 'label']].apply(lambda x: g.text(*x), axis = 1)

    plt.show();

sim_plot(vis, labels = people)
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mAttributeError[0m                            Traceback (most recent call last)
Cell [0;32mIn[15], line 18[0m
[1;32m     14[0m         to_label[[[38;5;124m'[39m[38;5;124mx[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124my[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124mlabel[39m[38;5;124m'[39m]][38;5;241m.[39mapply([38;5;28;01mlambda[39;00m x: g[38;5;241m.[39mtext([38;5;241m*[39mx), axis [38;5;241m=[39m [38;5;241m1[39m)
[1;32m     16[0m     plt[38;5;241m.[39mshow();
[0;32m---> 18[0m [43msim_plot[49m[43m([49m[43mvis[49m[43m,[49m[43m [49m[43mlabels[49m[43m [49m[38;5;241;43m=[39;49m[43m [49m[43mpeople[49m[43m)[49m

Cell [0;32mIn[15], line 10[0m, in [0;36msim_plot[0;34m(data, hue, labels, n_colors)[0m
[1;32m      4[0m pal [38;5;241m=[39m sns[38;5;241m.[39mcolor_palette([38;5;124m'[39m[38;5;124mcolorblind[39m[38;5;124m'[39m, n_colors [38;5;241m=[39m n_colors) [38;5;28;01mif[39;00m hue [38;5;28;01melse[39;00m [38;5;28;01mNone[39;00m
[1;32m      5[0m g [38;5;241m=[39m sns[38;5;241m.[39mscatterplot(
[1;32m      6[0m     x [38;5;241m=[39m [38;5;124m'[39m[38;5;124mx[39m[38;5;124m'[39m, y [38;5;241m=[39m [38;5;124m'[39m[38;5;124my[39m[38;5;124m'[39m
[1;32m      7[0m     , hue [38;5;241m=[39m hue, palette [38;5;241m=[39m pal, alpha [38;5;241m=[39m [38;5;241m0.8[39m
[1;32m      8[0m     , data [38;5;241m=[39m data, ax [38;5;241m=[39m ax
[1;32m      9[0m )
[0;32m---> 10[0m [43mg[49m[38;5;241;43m.[39;49m[43mdespine[49m()
[1;32m     12[0m [38;5;28;01mif[39;00m labels:
[1;32m     13[0m     to_label [38;5;241m=[39m data[data[[38;5;124m'[39m[38;5;124mlabel[39m[38;5;124m'[39m][38;5;241m.[39misin(labels)]

[0;31mAttributeError[0m: 'Axes' object has no attribute 'despine'

